clippy_lints/src/matches.rs

```diff
@@ -200,7 +200,8 @@ fn check_single_match_opt_like(cx: &LateContext, ex: &Expr, arms: &[Arm], expr:
             }
             path.to_string()
         }
-        PatKind::Ident(BindByValue(MutImmutable), ident, None) => ident.node.to_string(),
+        PatKind::Binding(BindByValue(MutImmutable), ident, None) => ident.node.to_string(),
+        PatKind::Path(ref path) => path.to_string(),
```

Comment:
Is `Binding` really necessary here? My understanding was that variant were now resolved (and we only care about some known paths).